src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -40,6 +44,33 @@
 
 logger = logging.getLogger(__name__)
 
+# @TODO: Find a home for these
+
+from dataclasses import dataclass  # noqa: E402
+
+from pants.engine.target import AllTargets, Target  # noqa: E402
+
+
+@dataclass(frozen=True)
+class AllResourcesTargets:
+    resources: tuple[Target, ...]
+    files: tuple[Target, ...]
+
+
+@rule(desc="Find all resources  in project")
+def find_all_resources(all_targets: AllTargets) -> AllResourcesTargets:
+    resources = []
+    files = []
+    for tgt in all_targets:
+        if isinstance(tgt, ResourceTarget):
+            resources.append(tgt)
+        if isinstance(tgt, FileTarget):
+            files.append(tgt)
```

Comment:
I think we usually prefer to look at the fields, rather than the target type.
> Often, you may want to see if a target type has a particular Field registered. This is useful to filter targets. Use the methods .has_field() and .has_fields().
From https://www.pantsbuild.org/v2.9/docs/rules-api-and-target-api

So in this case, look for targets that has the `ResourceSourceField` and `FileSourceField` respectively. That way, you won't exclude valid other potentially valid targets..